demos/LowAllocationWebServer/HttpHeaders.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.Text.Utf8;
+using LowAllocationServer;
+
+namespace System.Net.Http.Buffered
+{
+    public struct HttpHeaders : IEnumerable<KeyValuePair<Utf8String, Utf8String>>
+    {
+        private readonly ByteSpan _bytes;
+        
+        public HttpHeaders(ByteSpan bytes)
+        {
+            _bytes = bytes;            
+        }                
+
+        public Utf8String this[string headerName]
```

Comment:
We could, but our server is not asking for it at the movement. But even if we do, we need to choose in what base we want to work: bytespan or utf8string.